src/sdl2_ui.cpp

```diff
@@ -153,7 +153,9 @@ uint32_t Sdl2Ui::GetTicks() const {
 }
 
 void Sdl2Ui::Sleep(uint32_t time) {
-#ifndef EMSCRIPTEN
+#ifdef EMSCRIPTEN
+	emscripten_sleep(time);
+#else
 	SDL_Delay(time);
```

Comment:
the docs say SDL_Delay is supposed to work, but it freezes the browser when I use.

Reply:
How about `std::this_thread::sleep_for`? The clock PR switches to that. We could add it earlier if needed for emscripten.

Reply:
Possible but you have to tell me the mangled name of this function. Must be added to the asyncify list.

Reply:
Somehow it works now with SDL_Delay so I keep this untouched for now and just removed the ifdef.
sleep_for also works, tested it but you already make this change in your PR.